moseq2_model/train/util.py

```diff
@@ -157,6 +154,47 @@ def get_labels_from_model(model):
     return labels
 
 
+def apply_model(model, whitening_params, data_dict, metadata, whiten='all'):
+    '''
+    Apply pre-trained model to data_dict. Note that this function might produce unexpected behavior
+    if the model was trained using separate transition matrices for different groups of sessions.
+    
+    Args:
+        model (ARHMM): pre-trained model
+        whitening_params (namedtuple or dict): whitening parameters
+        data_dict (OrderedDict): data to apply model to
+        metadata (dict): metadata for data_dict
+    
+    Returns:
+        labels (dict): dictionary of labels predicted per session after modeling
+    '''
+
+    # whiten data function
+    try:
+        mu, L, offset = whitening_params['mu'], whitening_params['L'], whitening_params['offset']
+        apply_whitening = lambda x: np.linalg.solve(L, (x-mu).T).T + offset
+    except:
+        print('Whitening parameters not found.')
```

Comment:
Don't use an `except` block without explicitly mentioning any form of exception, it's bad practice.

Also, the rest of this function will error out, because `apply_whitening` is not defined, so this is incorrectly handled.

What use case are you trying to handle here?

Reply:
Yeah we should check before this function if `whitened_parameters` exist

Reply:
You don't want to catch a `KeyboardInterrupt` exception so whenever you use `except` you must at the very least catch the `Exception` class, which does not encompass the` KeyboardInterrupt` exception.

Reply:
I meant `KeyError` from indexing a key that is not actually in the dictionary.

Reply:
try...except and whitening parameters check
37fb3ca0dcb6923a3ea604cf82ec5cb7620032d1
8e6b27857d4bfd94e4ef0ee27073a866368b73da

adding PC_file and model paths:
1e0f6c78458a1bbc6901c2a03ad09a3dc60ca611
b1edb98e413ac686d7bd89ab44f94ec51defbae3